Formula/traefik.rb

```diff
@@ -0,0 +1,37 @@
+require "language/go"
+
+class Traefik < Formula
+  desc "Træfik, a modern reverse proxy"
+  homepage "https://traefik.io"
+  url "https://github.com/containous/traefik/archive/v1.3.7.tar.gz"
+  sha256 "dcea59508281c6c286c2fe81065cf3af3eeba1901f196588c773f9789f5fa9c8"
+  head "https://github.com/containous/traefik.git"
+
+  depends_on "go" => :build
+
+  go_resource "github.com/jteeuwen/go-bindata" do
+    url "https://github.com/jteeuwen/go-bindata.git",
+        :revision => "a0ff2567cfb70903282db057e799fd826784d41d"
+  end
+
+  def install
+    ENV["GOPATH"] = buildpath
+    dir = buildpath/"src/github.com/containous/traefik"
+    dir.install buildpath.children
+    ENV.prepend_create_path "PATH", buildpath/"bin"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    cd "src/github.com/jteeuwen/go-bindata/go-bindata" do
+      system "go", "install"
+    end
+
+    cd dir do
+      system "go", "generate"
+      system "go", "build", "-o", bin/"traefik", "./cmd/traefik"
+    end
+  end
+
+  test do
+    system bin/"traefik", "version"
```

Comment:
The test will need to exercise and verify the functionality of the software once it has been installed, not just run the version command.